### PR TITLE
Add docs for Action, ActionCamera, ActionCatmullRom, ActionGrid, Animati...

### PR DESCRIPTION
**Pull request by SuYaoHui**
...on, AnimationCache.

### CODE REVIEW

File: cocos/2d/CCAction.h [hunk 1]
```diff
@@ -53,58 +53,94 @@ class CC_DLL Action : public Ref, public Clonable
...
+     * @param target A certain target.
+     */
     virtual void startWithTarget(Node *target);
 
     /** 
-    called after the action has finished. It will set the 'target' to nil.
-    IMPORTANT: You should never call "[action stop]" manually. Instead, use: "target->stopAction(action);"
-    */
+     * Called after the action has finished. It will set the 'target' to nil.
+     * IMPORTANT: You should never call "[action stop]" manually. Instead, use: "target->stopAction(action);".
```
Comment (minggo) [inline cocos/2d/CCAction.h]: [action stop] -> Action::stop()
[action stop] is object-c grammar

File: cocos/2d/CCAction.h [hunk 2]
```diff
@@ -285,27 +347,27 @@ class CC_DLL Follow : public Action
     /**
      * Initializes the action with a set boundary or with no boundary.
      *
-     * @param followedNode  The node to be followed.
-     * @param rect  The boundary. If \p rect is equal to Rect::ZERO, it'll work
+     * @param FollowedNode  The node to be followed.
+     * @param Rect  The boundary. If \p rect is equal to Rect::ZERO, it'll work
```
Comment (minggo) [inline cocos/2d/CCAction.h]: wrong parameter names

File: cocos/2d/CCAction.h [hunk 3]
```diff
@@ -239,14 +301,14 @@ class CC_DLL Follow : public Action
     /**
      * Creates the action with a set boundary or with no boundary.
      *
-     * @param followedNode  The node to be followed.
-     * @param rect  The boundary. If \p rect is equal to Rect::ZERO, it'll work
+     * @param FollowedNode  The node to be followed.
+     * @param Rect  The boundary. If \p rect is equal to Rect::ZERO, it'll work
```
Comment (minggo) [inline cocos/2d/CCAction.h]: Wrong parameter names.

File: cocos/2d/CCActionCatmullRom.h [hunk 4]
```diff
@@ -73,47 +74,67 @@ class CC_DLL PointArray : public Ref, public Clonable
...
+    /** initializes a Catmull Rom config with a capacity hint.
+     *
      * @js NA
+     * @param capacity The size of the array.
+     * @return True.
      */
     bool initWithCapacity(ssize_t capacity);
 
-    /** appends a control point
+    /** appends a control point.
```
Comment (minggo) [inline cocos/2d/CCActionCatmullRom.h]: appends -> Appends.
The same issue in many places.

File: cocos/2d/CCActionCatmullRom.h [hunk 5]
```diff
@@ -73,47 +74,67 @@ class CC_DLL PointArray : public Ref, public Clonable
...
+     * @param index replace the point to array in index.
      */
     void replaceControlPoint(Vec2 &controlPoint, ssize_t index);
 
-    /** get the value of a controlPoint at a given index
+    /** get the value of a controlPoint at a given index.
+     *
      * @js NA
+     * @param index Get the point in index.
+     * @return A PointArray.
```
Comment (minggo) [inline cocos/2d/CCActionCatmullRom.h]: It returns a `Vec2`

File: cocos/2d/CCAnimation.h [hunk 6]
```diff
@@ -131,89 +155,127 @@ class CC_DLL AnimationFrame : public Ref, public Clonable
...
+    /** Gets the array of AnimationFrames.
+     * 
+     * @return The array of AnimationFrames.
+     */
     const Vector<AnimationFrame*>& getFrames() const { return _frames; };
     
-    /** Sets the array of AnimationFrames */
+    /** Sets the array of AnimationFrames. 
+     *
+     * @param The array of AnimationFrames.
```
Comment (minggo) [inline cocos/2d/CCAnimation.h]: What's the parameter?

File: cocos/2d/CCAnimation.h [hunk 7]
```diff
@@ -131,89 +155,127 @@ class CC_DLL AnimationFrame : public Ref, public Clonable
...
+    /** Gets the times the animation is going to loop. 0 means animation is not animated. 1, animation is executed one time, ... 
+     *
+     * @return The times the animation is going to loop.
+     */
     unsigned int getLoops() const { return _loops; };
     
-    /** Sets the times the animation is going to loop. 0 means animation is not animated. 1, animation is executed one time, ... */
+    /** Sets the times the animation is going to loop. 0 means animation is not animated. 1, animation is executed one time, ... 
+     *
+     * @param the times the animation is going to loop.
```
Comment (minggo) [inline cocos/2d/CCAnimation.h]: What's the parameter?

**Comment by IdVincentYang** [comment]
[ci skip]

**Pull request closed without merging**